utility/reactive-query/src/reactive_query/operator/fork/internal.rs

```diff
@@ -219,16 +219,10 @@ pub struct ForkComputeView<T: QueryCompute> {
 impl<Map: QueryCompute> ForkComputeView<Map> {
   pub fn resolve(&mut self, cx: &QueryResolveCtx) -> ForkedView<Map::View> {
     // early return if view has already computed
-    {
-      let resolved = self.view_resolve.read();
-      let resolved: &Option<Weak<dyn Any + Send + Sync>> = &resolved;
-      if let Some(view) = resolved {
-        if let Some(view) = view.upgrade() {
-          return ForkedView {
-            inner: view.downcast::<Map::View>().unwrap(),
-          };
-        }
-      }
+    if let Some(view) = self.already_resolved_view.as_ref() {
```

Comment:
The `already_resolved_view` is only used as the strong view holder at the compute struct creation time, and it is **not** shared among the multiple compute structs. The `view_resolve` is shared, so I think we should still check that.